spinn_front_end_common/interface/interface_functions/front_end_common_locate_executable_start_types.py

```diff
@@ -0,0 +1,38 @@
+from spinn_front_end_common.utilities import exceptions
+from spinn_utilities.progress_bar import ProgressBar
+
+
+class FrontEndCommonLocateExecutableStartType(object):
+
+    def __call__(
+            self, placements, graph, executable_finder, graph_mapper=None):
```

Comment:
I suspect this algorithm doesn't need all these inputs - it could probably operate simply on the application / machine graph.